--- origsrc/phc-winner-argon2-20171227/Makefile	2017-12-27 13:56:31.000000000 -0600
+++ src/phc-winner-argon2-20171227/Makefile	2018-04-29 20:13:15.027783500 -0500
@@ -58,37 +58,39 @@ BUILD_PATH := $(shell pwd)
 KERNEL_NAME := $(shell uname -s)
 
 LIB_NAME=argon2
+LIB_PFX=lib
 ifeq ($(KERNEL_NAME), Linux)
-	LIB_EXT := so.$(ABI_VERSION)
+	LIB_EXT := .so.$(ABI_VERSION)
 	LIB_CFLAGS := -shared -fPIC -fvisibility=hidden -DA2_VISCTL=1
-	SO_LDFLAGS := -Wl,-soname,lib$(LIB_NAME).$(LIB_EXT)
+	SO_LDFLAGS := -Wl,-soname,lib$(LIB_NAME)$(LIB_EXT)
 	LINKED_LIB_EXT := so
 endif
 ifeq ($(KERNEL_NAME), $(filter $(KERNEL_NAME),FreeBSD NetBSD OpenBSD))
-	LIB_EXT := so
+	LIB_EXT := .so
 	LIB_CFLAGS := -shared -fPIC
 endif
 ifeq ($(KERNEL_NAME), Darwin)
-	LIB_EXT := $(ABI_VERSION).dylib
-	LIB_CFLAGS := -dynamiclib -install_name @rpath/lib$(LIB_NAME).$(LIB_EXT)
+	LIB_EXT := .$(ABI_VERSION).dylib
+	LIB_CFLAGS := -dynamiclib -install_name @rpath/lib$(LIB_NAME)$(LIB_EXT)
 	LINKED_LIB_EXT := dylib
 endif
 ifeq ($(findstring CYGWIN, $(KERNEL_NAME)), CYGWIN)
-	LIB_EXT := dll
-	LIB_CFLAGS := -shared -Wl,--out-implib,lib$(LIB_NAME).$(LIB_EXT).a
+	LIB_PFX = cyg
+	LIB_EXT := -$(ABI_VERSION).dll
+	LIB_CFLAGS := -shared -Wl,--out-implib,lib$(LIB_NAME).dll.a
 endif
 ifeq ($(findstring MINGW, $(KERNEL_NAME)), MINGW)
-	LIB_EXT := dll
-	LIB_CFLAGS := -shared -Wl,--out-implib,lib$(LIB_NAME).$(LIB_EXT).a
+	LIB_EXT := .dll
+	LIB_CFLAGS := -shared -Wl,--out-implib,lib$(LIB_NAME)$(LIB_EXT).a
 endif
 ifeq ($(findstring MSYS, $(KERNEL_NAME)), MSYS)
-	LIB_EXT := dll
-	LIB_CFLAGS := -shared -Wl,--out-implib,lib$(LIB_NAME).$(LIB_EXT).a
+	LIB_EXT := .dll
+	LIB_CFLAGS := -shared -Wl,--out-implib,lib$(LIB_NAME)$(LIB_EXT).a
 endif
 ifeq ($(KERNEL_NAME), SunOS)
 	CC := gcc
 	CFLAGS += -D_REENTRANT
-	LIB_EXT := so
+	LIB_EXT := .so
 	LIB_CFLAGS := -shared -fPIC
 endif
 
@@ -98,7 +100,7 @@ ifeq ($(CC), clang)
 endif
 endif
 
-LIB_SH := lib$(LIB_NAME).$(LIB_EXT)
+LIB_SH := $(LIB_PFX)$(LIB_NAME)$(LIB_EXT)
 LIB_ST := lib$(LIB_NAME).a
 
 ifdef LINKED_LIB_EXT
